League.py

```diff
@@ -0,0 +1,157 @@
+
+    data = json.loads(gameInfo.text)
+    found = False
+
+    for i in range(0, len(data["participantIdentities"])):
+        partInfo = data["participantIdentities"][i]
+        temp = partInfo["player"]["summonerName"]
+        if(temp == summonerName):
+            pid = partInfo["participantId"]
+            found = True
+            break;
+    if (found == False):
+        print("There are no games for this summoner")
+
+    for i in range(0, len(data["participants"])):
+        partData = data["participants"][i]
+        partID = partData["participantId"]
+
+        if(partID == pid):
+            partGameInfo = data["participants"][pid-1]
+            statsDict = {'winLose':partGameInfo["stats"]["win"], 
+            'kills': partGameInfo["stats"]["kills"],
+            'deaths' : partGameInfo["stats"]["deaths"],
+            'assists': partGameInfo["stats"]["assists"],
+            'role' : partGameInfo["timeline"]["role"],
+            'lane' : partGameInfo["timeline"]["lane"],
+            'champion' : partGameInfo["championId"],
+            'gameMode' : data["gameMode"],
+            'champInfo' : getChampionInfo(partGameInfo["championId"])
+            }
+
+    return statsDict
+
+def getChampionInfo(championID):
+    with open('Champions.json','r', encoding='utf-8') as json_file:
+        championData = json.loads(json_file.read())
+    for i in championData["data"]:
+        if(str(championID) == str(championData["data"][i]["key"])):
+            # print(championData["data"][i]["image"]["full"])
+            return(championData["data"][i]["image"]["full"])
+    
+            
+def printOutput(statsList):
+
+    for i in statsList: 
+        if(i['winLose'] == True):
+            print("Win")
+        else: 
+            print("Defeat")
+        print("KDA:", i['kills'], i['deaths'], i['assists'])
+        print("Role:", i['role'])
+        print("Lane:", i['lane'])
+        print("Champion:", i['champion'])
+        print("Game Mode:", i['gameMode'])
+        print()
+        
+def getStatsofGames(summonerName, numberMatches, matchInfo, apiKey):
+
+    print(summonerName + "'s last " + str(numberMatches) + " games")
+    print()
+    statsList = []
+
+    for i in range(numberMatches):
+        match = matchInfo["matches"][i]
+        matchid = match["gameId"]
+        gameUrl = "https://na1.api.riotgames.com/lol/match/v4/matches/"+str(matchid)+"?api_key="+apiKey
+        gameInfo = requests.get(gameUrl)
+        statsDict = getStats(gameInfo, summonerName)
+        statsList.append(statsDict.copy())
+
+    return statsList
+
+# def main():
+#     apiKey = getAPIKey()
+#     summonerName, numberMatches = getUserInput()
+#     accountid = getAccountId(summonerName, apiKey)
+#     if accountid != None:
+#         matchInfo = getMatches(accountid, apiKey)
+#         statsList = getStatsofGames(summonerName, numberMatches, matchInfo, apiKey)
+#         printOutput(statsList)
+
+@app.route('/set', methods=['POST'])
```

Comment:
You should seperate app/routing flask methods from your logic code above into different files/classes